services/claudie-operator/pkg/controller/controller.go

```diff
@@ -55,7 +58,61 @@ func (r *InputManifestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 				log.Error(err, "secret not found", "will try again in", REQUEUE_AFTER_ERROR, "name", n.SecretRef.Name, "namespace", n.SecretRef.Namespace)
 				return ctrl.Result{RequeueAfter: REQUEUE_AFTER_ERROR}, nil
 			}
+
 			snwd.Endpoint = n.Endpoint
+			snwd.Username = "root"
+			if n.Username != "" {
+				snwd.Username = n.Username
+
+				// Parse private key.
+				pk, err := ssh.ParsePrivateKey(snwd.Secret.Data["privatekey"])
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Failed to parse private key for %s.", snwd.Endpoint))
+					return ctrl.Result{RequeueAfter: REQUEUE_AFTER_ERROR}, nil
+				}
+
+				// Define the SSH client configuration.
```

Comment:
I think that this is a bad idea to implement the validation here. If you do so, the connection will be established each time the reconciliation goes on, so it can go on every few seconds. 

I think it will be better to implement this in ansibler, as that service is responsible for the OS layer tasks

Reply:
I agree. I hadn't realized that it would establish the connection on every reconciliation. I just wanted to perform this verification as soon as possible to give users fast feedback. Is there also any other way to implement this? Besides the current one and the `ansibler` one you proposed.

Reply:
I think that ansibler is the earliest you can get for static nodes. Terraformer doesn't interact with static nodes, so ansibler is the next one. Maybe @Despire or @bernardhalas could share an opinion on that.

Reply:
Maybe one more thing to note. How will the user reconcile, if this check fails in the `ansibler`? Does changing the username attribute trigger the reconciliation?

Reply:
Similary like any other attribute. The workflow would have to fail/finnish, and then the changes will be applied.

Reply:
Okay, and to provide faster feedback to the user what do you think about performing this check in `builder`? Is it possible?

Reply:
I would like to add it [here](https://github.com/berops/claudie/blob/20f48814a4e3378adaebedc63e32f667f67d1e92/services/builder/domain/usecases/workflow_helpers.go#L78), before sending request to the `terraformer`.

Reply:
I think ansibler would be the goto place to insert this kind of stuff. Otherwise we will end up with some validation distributed everywhere.

When you think about it. The default is root. If someone wants to use another username then he must be sure it has root privileges. otherwise setup up the node will fail.

Reply:
> The default is root. If someone wants to use another username then he/she must be sure it has root privileges.

In that case, should we even perform the checks?

Reply:
I believe not since the playbooks will fail.

Reply:
That's true, but it will take about ~30 minutes to fail, because of this [task](https://github.com/berops/claudie/blob/20f48814a4e3378adaebedc63e32f667f67d1e92/services/ansibler/server/ansible-playbooks/wireguard.yml#L8).

Reply:
Eventually, I decided to remove the check from the `claudie-operator` and don't implement it at all.